Sources/SwiftSyntax/BumpPtrAllocator.swift

```diff
@@ -0,0 +1,114 @@
+//===---------- BumpPtrAllocator.swift - Bump-pointer Allocaiton ----------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Bump-pointer allocation.
+struct BumpPtrAllocator {
```

Comment:
I think this ought to be a class. Each allocator has a distinct identity, and we want to be able to deallocate slabs when the last reference to the allocator is gone.

Reply:
Eh, I was planning to use this as a member of `SyntaxArena` that is a class. in #512 . But yeah, I will change this to class.